### PR TITLE
Add virtual Cortex-R8 platform

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**
Right now Zephyr does not support Cortex-R8 platforms.

This PR adds support for:

- architecture
- virtual SoC
- virtual board

No changes to external HALs are required.

The virtual target can be easily verified with Renode.

### CODE REVIEW

**Comment by PiotrZierhoffer** [comment]
Rebased as Renode version is bumped in the Zephyr CI and now includes Cortex-R8 support

File: boards/renode/cortex_r8_virtual/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Refer to the `Renode documentation <https://renode.readthedocs.io/en/latest/>`_
+to learn how to obtain Renode for your host.
+
+Programming and debugging
+*************************
+
+Building
+========
+
+Applications for the ``cortex_r8_virtual`` board configuration can be built
```
Comment (nordicjm) [inline boards/renode/cortex_r8_virtual/doc/index.rst:21]: *board target

**Review by nordicjm — COMMENTED** [review]
Board/SoC files OK, minor nit.

File: boards/renode/cortex_r8_virtual/doc/index.rst (hunk 1 above)
Comment (mateusz-holenko) [reply boards/renode/cortex_r8_virtual/doc/index.rst]: Thanks, applied the remark

**Review by ithinuel — CHANGES_REQUESTED** [review]
I find the name a bit too generic, would it be possible to name it `renode_cortex_r8` instead like there is for `qemu_*` targets ?

**Review by ithinuel — DISMISSED** [review]

**Comment by mateusz-holenko** [comment]
@nordicjm Addressed your remark. Is it fine from your perspective now?

File: boards/renode/cortex_r8_virtual/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
+.. _cortex_r8_virtual:
+
+Cortex-R8 Virtual
+#################
+
+Overview
+********
+
+The Cortex-R8 Virtual board is a virtual platform that can be emulated with Renode.
+Edit the ``cortex_r8_virtual.repl`` file to adapt the platform layout to your needs.
```
Comment (nordicjm) [inline boards/renode/cortex_r8_virtual/doc/index.rst:10]:
```suggestion
Edit the :zephyr_file:`boards/renode/cortex_r8_virtual/support/cortex_r8_virtual.repl` file to adapt the platform layout to your needs.
```

**Review by nordicjm — COMMENTED** [review]
Minor nit. (please use the `request re-review` button next to reviewer names in the top right, that way it gets added to their list of PRs needing reviews)

File: boards/renode/cortex_r8_virtual/doc/index.rst (hunk 2 above)
Comment (mateusz-holenko) [reply boards/renode/cortex_r8_virtual/doc/index.rst]: Changed as suggested.

**Review by ithinuel — APPROVED** [review]

**Comment by mateusz-holenko** [comment]
@fkokosinski could you take a look at this?

**Review by fkokosinski — APPROVED** [review]
LGTM from the perspective of the "Testing with Renode" area of maintenance which this SoC is covered by.

**Pull request merged**